File: back/Empresa.py
class Empresa:
    def __init__(self, db):
        self.db = db

    def add_empresa(self, nome_empresa, cnpj):
        try:
            self.db.cursor.execute(
                "INSERT INTO empresa (Nome, Cnpj) VALUES (%s, %s)", 
                (nome_empresa, cnpj)
            )
            self.db.conn.commit()
            return self.db.cursor.lastrowid
        except Exception as e:
            print(f"Erro ao adicionar empresa ao banco de dados: {e}")
            raise e
        
    def get_empresas(self):
        try:
            self.db.cursor.execute("SELECT * FROM empresa ORDER BY Nome")
            result = self.db.cursor.fetchall()
            return result
        except Exception as e:
            print(f"Erro ao obter empresas do banco de dados: {e}")
            raise e

    def delete_empresa(self, empresa_id):
        try:
            self.db.cursor.execute("DELETE FROM empresa WHERE ID = %s", (empresa_id,))
            self.db.conn.commit()
        except Exception as e:
            print(f"Erro ao deletar empresa do banco de dados: {e}")
            raise e

    def update_empresa(self, empresa_id, novo_nome, novo_cnpj):
        try:
            self.db.cursor.execute(
                "UPDATE empresa SET Nome = %s, Cnpj = %s WHERE ID = %s", 
                (novo_nome, novo_cnpj, empresa_id)
            )
            self.db.conn.commit()
        except Exception as e:
            print(f"Erro ao atualizar empresa no banco de dados: {e}")
            raise e
        
    def empresa_avaliacao_insert(self, avaliacao_id, empresa_id):
        try:
            self.db.cursor.execute("UPDATE avaliacao set ID_Empresa = %s WHERE ID = %s", (empresa_id, avaliacao_id))
            self.db.conn.commit()
        except Exception as e:
            print(f"Erro ao deletar empresa do banco de dados: {e}")
            raise e
        
    
    def update_empresa_ajuste_avaliacao_inicial(self, id_empresa, nome):
        try:
            self.db.cursor.execute(
                "UPDATE empresa SET Nome = %s WHERE ID = %s", 
                (nome, id_empresa)
            )
            self.db.conn.commit()
        except Exception as e:
            print(f"Erro ao atualizar empresa no banco de dados: {e}")
            raise e

File: backjs/classes/Empresa.js
class Empresa {
    constructor(db) {
        this.db = db;
    }

    async addEmpresa(nomeEmpresa, cnpj) {
        const query = "INSERT INTO empresa (Nome, Cnpj) VALUES (?, ?)";
        try {
            const result = await this.db.executeQuery(query, [nomeEmpresa, cnpj]);
            return result.insertId; // Retorna o ID da nova empresa inserida
        } catch (err) {
            console.error("Erro ao adicionar empresa ao banco de dados:", err);
            throw err;
        }
    }

    async getEmpresas() {
        const query = "SELECT * FROM empresa ORDER BY Nome";
        try {
            const empresas = await this.db.fetchAll(query);
            return empresas;
        } catch (err) {
            console.error("Erro ao obter empresas do banco de dados:", err);
            throw err;
        }
    }

    async deleteEmpresa(empresaId) {
        const query = "DELETE FROM empresa WHERE ID = ?";
        try {
            await this.db.executeQuery(query, [empresaId]);
        } catch (err) {
            console.error("Erro ao deletar empresa do banco de dados:", err);
            throw err;
        }
    }

    async updateEmpresa(empresaId, novoNome, novoCnpj) {
        const query = "UPDATE empresa SET Nome = ?, Cnpj = ? WHERE ID = ?";
        try {
            await this.db.executeQuery(query, [novoNome, novoCnpj, empresaId]);
        } catch (err) {
            console.error("Erro ao atualizar empresa no banco de dados:", err);
            throw err;
        }
    }

    async empresaAvaliacaoInsert(avaliacaoId, empresaId) {
        const query = "UPDATE avaliacao SET ID_Empresa = ? WHERE ID = ?";
        try {
            await this.db.executeQuery(query, [empresaId, avaliacaoId]);
        } catch (err) {
            console.error("Erro ao inserir empresa na avaliação:", err);
            throw err;
        }
    }

    async updateEmpresaAjusteAvaliacaoInicial(idEmpresa, nome) {
        const query = "UPDATE empresa SET Nome = ? WHERE ID = ?";
        try {
            await this.db.executeQuery(query, [nome, idEmpresa]);
        } catch (err) {
            console.error("Erro ao atualizar empresa no banco de dados:", err);
            throw err;
        }
    }
}

module.exports = Empresa;
